scripts/add_electricity.py

```diff
@@ -246,14 +270,26 @@ def attach_load(n, regions, load, admin_shapes, countries, scale):
         gpd.read_file(regions).set_index("name").reindex(substation_lv_i)
     ).dropna(
         axis="rows")  # TODO: check if dropna required here. NaN shapes exist?
-    load_path = load
-    gegis_load = xr.open_dataset(load_path)
+
+    cwd_path = os.getcwd()  # path $ ~/pypsa-africa
+    load_paths = []
+    for continent in region_load:
+        load_path = os.path.join(
+            cwd_path,
+            "resources",
+            str(ssp),
+            str(prediction_year),
+            "era5_" + str(weather_year),
+            str(continent).capitalize() + ".nc",
+        )
+        load_paths.append(load_path)
```

Comment:
**Should be addressed**

I highly recommend that this happens outside the script and in the `Snakefile`; file dependencies should all be made explicit there.

There all the information to construct the file paths are already available (via the `config`), so you could probably use this function in a very similar manner in the `Snakefile`.

The script would then simply get the path or paths via `snakemake.input["loads"]` (like it did before) and then read through `xr.open_dataset(snakemake.input["loads"]`

Reply:
Done with https://github.com/pypsa-meets-africa/pypsa-africa/pull/275/commits/f593c9db5de8b9cf9d2627bc1d370a1b713eae8a